src/librustc_target/spec/nvptx64_nvidia_cuda.rs

```diff
@@ -0,0 +1,73 @@
+use spec::{LinkerFlavor, Target, TargetOptions, TargetResult, PanicStrategy, MergeFunctions};
+use spec::abi::Abi;
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        arch: "nvptx64".to_string(),
+        data_layout: "e-i64:64-i128:128-v16:16-v32:32-n16:32:64".to_string(),
+        llvm_target: "nvptx64-nvidia-cuda".to_string(),
+
+        target_os: "cuda".to_string(),
+        target_vendor: "nvidia".to_string(),
+        target_env: String::new(),
+
+        linker_flavor: LinkerFlavor::PtxLinker,
+
+        target_endian: "little".to_string(),
+        target_pointer_width: "64".to_string(),
+        target_c_int_width: "32".to_string(),
+
+        options: TargetOptions {
+            // The linker can be installed from `crates.io`.
+            linker: Some("rust-ptx-linker".to_string()),
+
+            // With `ptx-linker` approach, it can be later overriden via link flags.
+            cpu: "sm_20".to_string(),
+
+            // TODO(denzp): create tests for the atomics.
+            max_atomic_width: Some(64),
```

Comment:
BTW I did a few basic tests, and it turns out the the `atomic_xadd` rust-intrinsic seems to already generate correct PTX when doing atomic adds on `u32` and `u64`. Hurray! I didn't test too many other atomic ops or sizes though.

(Sadly, `atomic_xadd` only supports integers and doesn't compile for `f32` and `f64` types.)